libs/langchain/langchain/text_splitter.py

```diff
@@ -261,15 +261,21 @@ async def atransform_documents(
 class CharacterTextSplitter(TextSplitter):
     """Implementation of splitting text that looks at characters."""
 
-    def __init__(self, separator: str = "\n\n", **kwargs: Any) -> None:
+    def __init__(
+        self, separator: str = "\n\n", is_separator_regex: bool = False, **kwargs: Any
+    ) -> None:
         """Create a new TextSplitter."""
         super().__init__(**kwargs)
         self._separator = separator
```

Comment:
should we just perform escaping here?

Reply:
This will affect the output when `keep_separator=True` because it is using `self._separator` to add the [separator back](https://github.com/langchain-ai/langchain/pull/7933/files#diff-d36ce27aa411821eb95a4e6d713a0fadfdd480a4dc16abb8fe71e82c97d87561R279) and will add the escaped separator. 
Another approach can be to save the escaped separator as well as the original separator. WDYT?